In [6]:
# time column

import csv
d=dict()
job_failed=0
job_passed=0
with open('Job_Events.csv') as csv_file:
    csv_reader = csv.reader(csv_file,delimiter=',')
    for row in csv_reader:
        p=row[0]
        
        if p not in d:d[p]=1
        else:d[p]+=1
    for key,val in list(d.items()):
        if key=='0':job_failed=val
        else:
            job_passed+=1
    print("No of times job passed successful",job_passed)
    print("No of times job failed",job_failed)

No of times job passed successful 2730
No of times job failed 7970


The column 1 of the job_events file contains time 
This column is mandatory field. 
The format of field is integer.

In [10]:
# missing info column
d=dict()
c=0
with open('Job_Events.csv') as csv_file:
    csv_reader = csv.reader(csv_file,delimiter=',')
    for row in csv_reader:
        p=row[1]
        if p not in d:d[p]=1
        else:d[p]+=1
    print(d)


{'': 10704}


The column 2 of the job_events file contains missing_info
This column is not mandatory field. 
The format of field is integer.

In [13]:
# job_id
d=dict()
c=0
with open('Job_Events.csv') as csv_file:
    csv_reader = csv.reader(csv_file,delimiter=',')
    for row in csv_reader:
        p=row[2]
        if p not in d:d[p]=1
        else:d[p]+=1
    for key,val in list(d.items()):
        c=c+1
    print("Total number of unique job_id are :",c)


Total number of unique job_id are : 4900


The column 3 of the job_events file contains job_id
This column is mandatory field. 
The format of field is integer.

In [15]:
# events_type
d=dict()
with open('Job_Events.csv') as csv_file:
    csv_reader = csv.reader(csv_file,delimiter=',')
    for row in csv_reader:
        p=row[3]
        if p not in d:d[p]=1
        else:d[p]+=1
    print(d)

{'0': 4900, '1': 4892, '4': 636, '5': 268, '3': 8}


The column 4 of the job_events file contains event_id
This column is mandatory field. 
The format of field is integer.
Machine events have value 0,1,2 which are diffrent from the value in job_events data.



In [32]:
# user column
d=dict()
lst=list()
c=0
with open('Job_Events.csv') as csv_file:
    csv_reader = csv.reader(csv_file,delimiter=',')
    for row in csv_reader:
        p=row[4]
        if p not in d:d[p]=1
        else:d[p]+=1
    for key,val in list(d.items()):
        lst.append((val,key))
    lst.sort(reverse=True)
    for val,key in lst:
        print(key,val)
        c=c+1
    print("Total number of unique users :",c)


mbrS6V2tRbksMPO0y8EI9B9XT2ASaPRTNndslCGN/2A= 1077
68CsCZngOkuQn5lVo+dNXKGoFx9ZmJXWR+W7NAUmEH4= 577
8OkMisyvwLRFV8wxW85Psk0o/my2wpLBMFuoz4pvotY= 552
fJYeclskJqPWsAT6TX/r9X5OiIZpSEb2PBGliYAOMxM= 460
F2+Gv53Pxd4KDRb/UsGECThH/XUOpcWKElUXJkhkt1c= 320
CwGnjPqF6z6mEbr5zJVXafRsbETt6gvdwvOX2l+GpsI= 284
nHinWYr2Nep8Mr79MMvmYCrGb0NdshyX6mlnZo/Ot+I= 186
ZpQmujQYX55FcN2RFvqqUkcz5z/Tovj2FaG8sFtlG48= 185
oPxcKd7feXmw+sZKrkp8/MvvNP/5kRG+Yy2fc/lz3Ks= 160
q6EjQj6yxdwUvvufuDKvfotZ1LoylbpNu4NguS48Lfo= 150
9bg757+k8xeWFApxOzOS5nnVAHYpBBQ+Pf1mEnsTZsQ= 147
ZHAzkawqKRJLjH0vstw83yLYltCctQ0DZEmZvQgnTvM= 132
r/Al6kYJOwZITr6wi4pAlEwyGv5TM2EkJ8woA5hszeA= 131
rNyxTd1B3RnDJBIofzoVksjnmaJEN+hW5W+yRHo2xsM= 126
70s3v5qRyCO/1PCdI6fVXnrW8FU/w+5CKRSa72xgcIo= 118
0r3naaX+ybfsUIahb35AvYPC6MarnPhbqkAtRJor0xE= 117
0yO12LxPGQxL+XkI4hS6Tlyx5V3KOvjHS0DQSOJeMbs= 116
QhKaMUnHYqtG1RJkmo+zqhY1dChfuOZ6oy8jYGDlsic= 107
dT4PmDV8BInbmhDgMzu5obSCBm763C/HqMpBHL96dBI= 104
HBL13RkuP8hPfuPHZHIJTQZkoEtBdUQ9gtpKwGmSfJc= 96
yJ9xqe3+iO6kAPKtwDfI

The column 5 of the job_events file contains user
This column is not mandatory field. 
The format of field is string_hash.
Total number of unique users : 521

In [37]:
# scheduling class
d=dict()
c=0
with open('Job_Events.csv') as csv_file:
    csv_reader = csv.reader(csv_file,delimiter=',')
    for row in csv_reader:
        p=row[5]
        if p not in d:d[p]=1
        else:d[p]+=1
    print(d)

{'3': 1885, '2': 3030, '1': 3610, '0': 2179}


The column 6 of the job_events file scheduling class
This column is not mandatory field. 
The format of field is integer.
Total number of unique schedulig class : 4

In [39]:
# job name
d=dict()
c=0
with open('Job_Events.csv') as csv_file:
    csv_reader = csv.reader(csv_file,delimiter=',')
    for row in csv_reader:
        p=row[6]
        if p not in d:d[p]=1
        else:d[p]+=1
    for key,val in list(d.items()):
        c=c+1
    print("total number of unique job name :",c)

total number of unique job name : 4235


The column 7 of the job_events file job name
This column is not mandatory field. 
The format of field is string_hash.
Total number of unique users : 4235

In [42]:
# logical job name
d=dict()
c=0
with open('Job_Events.csv') as csv_file:
    csv_reader = csv.reader(csv_file,delimiter=',')
    for row in csv_reader:
        p=row[7]
        if p not in d:d[p]=1
        else:d[p]+=1
    for key,val in list(d.items()):
        c=c+1
    print("Total number of logical job names :",c)

Total number of logical job names : 4092


The column 8 of the job_events file logical job name
This column is not mandatory field. 
The format of field is string_hash.
Total number of unique users : 4235